airflow/contrib/operators/gcp_function_operator.py

```diff
@@ -0,0 +1,571 @@
+
+    @staticmethod
+    def _validate_regexp(full_field_path, regexp, value):
+        # type: (str, str, str) -> None
+        if not re.match(regexp, value):
+            # Note matching of only the beginning as we assume the regexps all-or-nothing
+            raise FieldValidationException(
+                "The field '{}' value '{}' does not match regexp {}.".
+                format(full_field_path, value, regexp))
+
+    def _validate_dict(self, children_validation_specs, full_field_path, value):
+        # type: (dict, str, dict) -> None
+        for child_validation_spec in children_validation_specs:
+            self._validate_field(validation_spec=child_validation_spec,
+                                 dictionary_to_validate=value,
+                                 parent=full_field_path)
+        # Possibly we find out that we should not validate extra fields
+        # for future compatibility ?
+        # But this is super helpful to detect typos early,
+        # so we might want to leave it here. It's super easy
+        # To add future validations
+        # (and we might want to disable this check with some flag)
+        for field_name in value.keys():
+            if field_name not in [spec['name'] for spec in children_validation_specs]:
+                self.log.warning(
+                    "The field '{}' is present in the dictionary but "
+                    "is missing in validation specification '{}'".format(
+                        self._get_field_name_with_parent(field_name, full_field_path),
+                        children_validation_specs))
+
+    def _validate_union(self, children_validation_specs, full_field_path,
+                        dictionary_to_validate):
+        # type: (dict, str, dict) -> None
+        field_found = False
+        found_field_name = None
+        for child_validation_spec in children_validation_specs:
+            # Forcing optional so that we do not have to type optional = True
+            # in specification for all union fields
+            new_field_found = self._validate_field(
+                validation_spec=child_validation_spec,
+                dictionary_to_validate=dictionary_to_validate,
+                parent=full_field_path,
+                force_optional=True)
+            field_name = child_validation_spec['name']
+            if new_field_found and field_found:
+                raise FieldValidationException(
+                    "The field '{}' and '{}' are part of '{}' union so they are mutually "
+                    "exclusive. They should not be both present in the same dictionary!".
+                    format(field_name, found_field_name, full_field_path))
+            if new_field_found:
+                field_found = True
+                found_field_name = field_name
+        if not field_found:
+            self.log.warning(
+                "None of available variants of '{}' union was found in dictionary {}!. "
```

Comment:
Ditto here.

Reply:
Fixed in 3722a2be480c6d8080fc60491e8fa431b7eb6590